### PR TITLE
ETL: Download/Extract ZIP

### LINKED ISSUE #460: ETL: Translate Ruby/Capybara zip download into Python/Splinter
We need to be able to process the emails once received in the AWS bucket. The code for this is already written in Ruby/Capybara, we just have to translate to Python/Splinter.

- [ ] Download emails from S3
- [ ] Log in to CRIS
- [ ] Download the zip file
- [ ] Unzip the files

### PR DESCRIPTION
**Pull request by sergiogcx**
Closes #460 

How to test this script:
1. Update your staging and production env files, there are two new variables needed there.
2. Run: `source runetl.sh`
3. Run: `runetl build`
4. Run: `runetl [path to staging env file] app/process_cris_request_download.py`

What to expect:
In your `tmp/` folder you should see some files like these:
Screen Shot 2019-11-20 at 5 22 18 PM

1. The `email_` are raw emails stored in s3 and are now downloaded to the folder for processing.

2. If you see zip files, that means the download process succeeded.

3. Check your `data/` folder, if you see csv files there, it means the extraction of those files succeeded.

** OPTIONAL
4. Check the finished folder in s3, [you should see three folders](https://s3.console.aws.amazon.com/s3/buckets/atd-vision-zero-editor/txdot-email-finished/?region=us-east-1&tab=overview).

### CODE REVIEW

File: atd-etl/app/process/helpers_request_download.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
+"""
+CRIS - Request Download Helper
+Author: Austin Transportation Department, Data and Technology Office
```
Comment (mateoclarke) [inline atd-etl/app/process/helpers_request_download.py]: `Author: Austin Transportation Department, Data and Technology Services`

File: atd-etl/app/process/helpers_request_download.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    return "email_" + ntpath.basename(file_key)
+
+
+def download_s3_file(file_key):
+    """
+    Downloads an email file from s3
+    :param file_key: the full path to the email file
+    :return:
+    """
+    with open("/app/tmp/%s" % (get_file_name(file_key)), 'wb') as f:
```
Comment (mateoclarke) [inline atd-etl/app/process/helpers_request_download.py]: what does the `wb` mean here?

Comment (mateoclarke) [reply atd-etl/app/process/helpers_request_download.py]: and the syntax of the `%`?

File: atd-etl/app/process/helpers_request_download.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    return mailparser.parse_from_file("/app/tmp/%s" % (get_file_name(file_key)))
+
+
+def extract_email_download_link(email_body):
+    """
+    Returns the download url from an email text body.
+    :param email_body: full raw body from the file
+    :return: string
+    """
+    download_link_string = re.search(AWS_CRIS_REQUEST_DOWNLOAD_LINK_PATTERN, email_body, re.IGNORECASE)
```
Comment (mateoclarke) [inline atd-etl/app/process/helpers_request_download.py]: 💅

Comment (mateoclarke) [reply atd-etl/app/process/helpers_request_download.py]: learning about re and `.match` vs. .`group` methods

File: atd-etl/app/process/helpers_request_download.py (hunk 1 above)
Comment (sergiogcx) [reply atd-etl/app/process/helpers_request_download.py]: Will refactor that :/

File: atd-etl/app/process/helpers_request_download.py (hunk 2 above)
Comment (sergiogcx) [reply atd-etl/app/process/helpers_request_download.py]: wb = write binary, %s = string, % = replace %s with variable

**Pull request merged**